### ISSUE TITLE
An infinite loop with unresolved collisions

### ISSUE THREAD

**Issue by bartoszpankratz** [tolcz/ABCDeGraphGenerator.jl#2]
Hi,

I have found another error. In some cases, when code gets some unresolved collisions, it freezes and stuck in an infinite loop. Here is the package with an example: 
[infloop.zip](https://github.com/tolcz/ABCDeGraphGenerator.jl/files/6955170/infloop.zip)

**Comment by tlcz** [comment]
Hi @bartoszpankratz 
Thank you for pointing out yet another deficiency. Yes indeed, with the configuration you provided `config_model()` sometimes gets stuck in an infinite loop trying to resolve collisions in a global/background graph. I have two comments however:
1. the configuration you provided is an 'edge' case with very small fraction of edges falling into a global/background graph - I am not sure of its relevance for real-life scenarios
```
xi = "0.0001"                    # fraction of edges to fall in background graph
mu = "9.339997092e-5"            # mixing parameter
```
2. the error is 'inherited' and can be reproduced with the upstream base [bkamins/ABCDGraphGenerator.jl](https://github.com/bkamins/ABCDGraphGenerator.jl). It reappears in ABCDe as the same collision resolution logic is used in both. The fix needs to be sanctioned by the upstream owner @bkamins.

**Comment by bkamins** [comment]
https://github.com/bkamins/ABCDGraphGenerator.jl/pull/10 should fix it. can you please test and report there if now things are OK?

**Comment by bartoszpankratz** [comment]
@tolcz 

Tomasz, I have tested bkamins#10  and everything works fine. Could you include this fix (or confirm that it's already done)  in the ABCDe package?  I want to pursue the speed benchmark and I need the working parallel version.

Thanks!

**Comment by tlcz** [comment]
Hi @bartoszpankratz 
fixed by 7a76d830c6970ad82e2d93c86429f4338f24c765
Thank you.
